schema.ddl

```diff
@@ -73,3 +144,92 @@ ON subnet (provider_id);
 
 CREATE UNIQUE INDEX IF NOT EXISTS idx_subnet_id
 ON subnet (id);
+
+/*
+ * Model
+ */
+CREATE TABLE IF NOT EXISTS model (
+	uuid					TEXT PRIMARY KEY,
+	id						TEXT NOT NULL UNIQUE,
+	name					TEXT,
+	type					TEXT,
+	life					INT,
+	owner					TEXT,
+	migration_mode			TEXT,
+	environ_version			INT,
+	last_available_tools	TEXT,
+	meter_status			TEXT,
+	password_hash			TEXT,
+	force_destroyed			TEXT,
+	destroy_timeout			TEXT
+	-- cloud
+	-- cloud_region
+	-- cloud_credential
+	-- controller_id
+	-- sla
+);
+
+CREATE UNIQUE INDEX IF NOT EXISTS idx_model_id
+ON model (id);
+
+/*
+ * Application
+ */
+CREATE TABLE IF NOT EXISTS application (
+	uuid					TEXT PRIMARY KEY,
+	id						TEXT NOT NULL UNIQUE,
+	name					TEXT,
+	series					TEXT,
+	subordinate				BOOLEAN,
+	charm_url				TEXT,
+	channel					TEXT,
+	charm_origin			TEXT,
+	charm_modified_version	INT,
+	force_charm				BOOLEAN,
+	life					INT,
+	min_units				INT,
+	exposed					BOOLEAN,
+	desired_scale			INT,
+	password_hash			TEXT,
+	placement				TEXT,
+	has_resources			BOOL
+	-- unit_count
+	-- relation_count
+	-- tools
+	-- metric_credentials
+	-- exposed_endpoints
+);
+
+CREATE UNIQUE INDEX IF NOT EXISTS idx_application_id
+ON application (id);
+
+CREATE TABLE IF NOT EXISTS application_offer (
+	uuid			TEXT PRIMARY KEY,
+	id				TEXT NOT NULL UNIQUE,
+	name			TEXT,
+	application_id	TEXT NOT NULL,
+	-- application_description
+	CONSTRAINT		fk_application_offer_application
+		FOREIGN KEY	(application_id)
+		REFERENCES	application(id)
+);
+
+CREATE UNIQUE INDEX IF NOT EXISTS idx_application_offer_id
+ON application_offer (id)
+
+CREATE TABLE IF NOT EXISTS endpoint_binding (
+	uuid			TEXT PRIMARY KEY,
+	id				TEXT NOT NULL UNIQUE,
```

Comment:
Don't think we'll need an ID here, but there will be a unique index on application_id and endpoint_name.

Reply:
Sure thing, I didn't want to include endpoint_name in the index, because it felt wrong to use something that wasn't an id.